skrub/_datetime_encoder.py

```diff
@@ -18,62 +21,494 @@
+        - Series: :class:`Series` of :class:`datetime64` dtype (or
+          :class:`Series` of :class:`object` dtype containing
+          :class:`datetime.datetime`)
+        - DataFrame: :class:`Series` of :class:`datetime64` dtype (or
+          :class:`Series` of :class:`object` dtype containing
+          :class:`datetime.datetime`)
+
+    Raises
+    ------
+    ParserError
+        When parsing a date from string fails.
+    ValueError
+        When another datetime conversion error happens. For example when one
+        of 'year', 'month', day' columns is missing in a :class:`DataFrame`, or
+        when a Timezone-aware :class:`datetime.datetime` is found in an array-like
+        of mixed time offsets, and ``utc=False``.
+
+    See Also
+    --------
+    :func:`pandas.to_datetime`
+    """
+    kwargs["errors"] = errors
+
+    # dataframe
+    if hasattr(X, "__dataframe__"):
+        return _to_datetime_dataframe(X, **kwargs)
+
+    # series, this attribute is available since Pandas 2.1.0
+    elif hasattr(X, "__column_consortium_standard__"):
+        return _to_datetime_series(X, **kwargs)
+
+    # 2d array
+    elif isinstance(X, Iterable) and np.asarray(X).ndim == 2:
+        X = _to_datetime_2d_array(np.asarray(X), **kwargs)
+        return np.vstack(X).T
+
+    # scalar or unknown type
+    return pd.to_datetime(X, **kwargs)
+
+
+def _to_datetime_dataframe(X, **kwargs):
+    """Dataframe specialization of ``_to_datetime_2d``.
+
+    Parameters
+    ----------
+    X : Pandas or Polars dataframe
+
+    Returns
+    -------
+    X : Pandas or Polars dataframe
+    """
+    _, px = get_df_namespace(X)
+    index = getattr(X, "index", None)
+    X_split = [X[col].to_numpy() for col in X.columns]
+    X_split = _to_datetime_2d(X_split, **kwargs)
+    X_split = {col: X_split[col_idx] for col_idx, col in enumerate(X.columns)}
+    X = pd.DataFrame(X_split, index=index)
+    # conversion is px is Polars, no-op if Pandas
+    return px.DataFrame(X)
+
+
+def _to_datetime_series(X, **kwargs):
+    """Series specialization of :func:`pandas.to_datetime`.
+
+    Parameters
+    ----------
+    X : Pandas or Polars series
+
+    Returns
+    -------
+    X : Pandas or Polars series
+    """
+    _, px = get_df_namespace(X.to_frame())
+    index = getattr(X, "index", None)
+    name = X.name
+    X = pd.to_datetime(X, **kwargs)
+    X = pd.Series(X, index=index, name=name)
+    # conversion is px is Polars, no-op if Pandas
+    return px.Series(X)
+
+
+def _to_datetime_2d_array(X, **kwargs):
+    """2d array specialization of ``_to_datetime_2d``.
+
+    Parameters
+    ----------
+    X : ndarray of shape ``(n_samples, n_features)``
+
+    Returns
+    -------
+    X_split : list of array, of shape ``n_features``
+    """
+    X_split = np.hsplit(X, X.shape[1])
+    X_split = [X_col.ravel() for X_col in X_split]
+    return _to_datetime_2d(X_split, **kwargs)
+
+
+def _to_datetime_2d(
+    X_split,
+    indices=None,
```

Comment:
it seems `indices` is always `None`? at least if only the documented **kwargs of `to_datetime` are used

Reply:
yep sorry for the noise!